crates/stages/src/stages/bodies.rs

```diff
@@ -222,18 +287,41 @@ impl<DB: Database, D: BodyDownloader> Stage<DB> for BodyStage<D> {
                 tx_block_cursor.delete_current()?;
             }
 
-            // Delete all transactions that belong to this block
-            for tx_id in block_meta.tx_num_range() {
```

Comment:
Is it possible to have transactions in both MDBX and snapshots provided you don't do a fresh sync? If so, what happens if we don't unwind MDBX? I.e. would the transactions in MDBX be the source of truth or the snapshot? If it's MDBX then we need to keep this for now

Reply:
what happens here:

> If so, what happens if we don't unwind MDBX? I.e. would the transactions in MDBX be the source of truth or the snapshot? If it's MDBX then we need to keep this for now

Reply:
the source of truth is MDBX. I think we need to check the highest snapshot here, and delete from the database between `highest_snapshot_block` and `tip`, and only then from the static files.